modules/transport-netty4/src/test/java/org/elasticsearch/http/netty4/Netty4HttpPipeliningHandlerTests.java

```diff
@@ -211,6 +215,43 @@ public void testPipeliningRequestsAreReleased() throws InterruptedException {
         }
     }
 
+    public void testSmallFullResponsesAreSentDirectly() {
+        final List<Object> messagesSeen = new ArrayList<>();
+        final var embeddedChannel = new EmbeddedChannel(capturingHandler(messagesSeen), getTestHttpHandler());
+        embeddedChannel.writeInbound(createHttpRequest("/test"));
+        final Netty4HttpRequest request = embeddedChannel.readInbound();
+        final var maxSize = (int) NettyAllocator.suggestedMaxAllocationSize() / 2;
+        final var content = new ZeroBytesReference(between(0, maxSize));
+        final var response = request.createResponse(RestStatus.OK, content);
+        final var promise = embeddedChannel.newPromise();
+        embeddedChannel.writeAndFlush(response, promise);
+        assertTrue(promise.isDone());
+        assertThat(messagesSeen, hasSize(1));
+        assertSame(response, messagesSeen.get(0));
```

Comment:
should we also assert that the seen message is an instance of full HTTP response?

Reply:
Yep can do, see 5d30ae9df72